modules/kafka/kafka-dest-driver.c

```diff
@@ -345,20 +459,33 @@ kafka_dd_init(LogPipe *s)
   if (self->kafka == NULL)
     {
       msg_error("kafka: error constructing kafka connection object, perhaps metadata.broker.list property is missing?",
-                evt_tag_str("topic", self->topic_name),
+                evt_tag_str("topic", self->topic_name->template),
                 evt_tag_str("driver", self->super.super.super.id),
                 log_pipe_location_tag(&self->super.super.super.super));
       return FALSE;
     }
 
-  self->topic = _construct_topic(self);
-  if (self->topic == NULL)
+  if (strchr(self->topic_name->template, '$') != NULL)
```

Comment:
Please move this check to a separate function called something like `_topic_name_is_a_template(self)`.